### PR TITLE
Updating Function primary script determination logic

### PR DESCRIPTION
**Pull request by mathewc**
This change is for the same change made in the Functions runtime (see https://github.com/Azure/azure-webjobs-sdk-script/pull/352).

The plan is for these two changes to go out together in the same release, otherwise things will get out of sync.

### CODE REVIEW

File: Kudu.Core/Functions/FunctionManager.cs [hunk 1]
```diff
@@ -262,64 +262,60 @@ public void DeleteFunction(string name)
...
-                return functionPath;
-            }
-            else if (functionFiles.Length == 1)
+            return DeterminePrimaryScriptFile(functionConfig, functionFiles);
+        }
+
+        // Logic for this function is copied from here:
+        // https://github.com/Azure/azure-webjobs-sdk-script/blob/dev/src/WebJobs.Script/Host/ScriptHost.cs
+        // These two implementations must stay in sync!
+        internal static string DeterminePrimaryScriptFile(JObject functionConfig, string[] functionFiles)
```
Comment (mathewc) [inline Kudu.Core/Functions/FunctionManager.cs]: This function was copied verbatim from the Functions repo. Unit tests for it are in that repo.

**Comment by davidebbo** [comment]
:shipit: But hold off on merging to master until we have a clear plan to get the new Functions Runtime out. Let's start this conversation.

**Comment by mathewc** [comment]
Yes, the Functions Runtime won't be ready for another release for at least a week or more. I won't check this in until we're ready on that side. Then we can coordinate the releases.

**Comment by davidebbo** [comment]
Rebased and merged.  81bebe79e7ba08c680b168aeb29326a255e9ef14

**Pull request closed without merging**